src/lib/shell/commands/Meta.cpp

```diff
@@ -55,12 +56,20 @@ static int VersionHandler(int argc, char ** argv)
     return 0;
 }
 
+static int FactoryResetHandler(int argc, char ** argv)
+{
+    streamer_printf(streamer_get(), "Performing factory reset ... \r\n");
+    DeviceLayer::ConfigurationMgr().InitiateFactoryReset();
+    return 0;
+}
+
 void RegisterMetaCommands()
 {
     static shell_command_t sCmds[] = {
         { &ExitHandler, "exit", "Exit the shell application" },
         { &HelpHandler, "help", "List out all top level commands" },
         { &VersionHandler, "version", "Output the software version" },
+        { &FactoryResetHandler, "factoryreset", "Performs device factory reset" },
```

Comment:
The command more suits the "config" or "device" command set. The meta commands mean commands related to the shell engine itself.

Reply:
In that case I added separate set of "device" commands